apps/modernization-ui/src/apps/page-builder/context/ConditionsContext.tsx

```diff
@@ -0,0 +1,70 @@
+/* eslint-disable camelcase */
+import { createContext, useState, Dispatch, SetStateAction } from 'react';
+
+interface ConditionContextData {
+    filter: any;
+    setFilter: (filter: any) => void;
+    searchQuery: string;
+    setSearchQuery: (query: string) => void;
+    currentPage: number;
+    setCurrentPage?: (page: number) => void;
+    sortBy: string;
+    setSortBy: (name: string) => void;
+    sortDirection: string;
+    setSortDirection: (direction: string) => void;
+    pageSize: number;
+    setPageSize: Dispatch<SetStateAction<number>>;
+    isLoading: boolean;
+    setIsLoading: (status: boolean) => void;
+}
+
+const conditionContextDefaultValue: ConditionContextData = {
+    filter: {},
+    setFilter: () => {},
+    searchQuery: '',
+    setSearchQuery: () => {},
+    currentPage: 1,
+    setCurrentPage: () => {},
+    sortBy: 'conditionShortNm',
+    setSortBy: () => {},
+    sortDirection: 'asc',
+    setSortDirection: () => {},
+    pageSize: 10,
+    setPageSize: () => {},
```

Comment:
Page size doesn't seem to be hooked up to the selector in the UI

Reply:
So this is kind of an issue, i have a fix pushed here for now but we may want to look into it... or not?  For every table we have created a context for the page size, current page etc etc.  well all these contexts are just copied and pasted from each other with only a couple of different minor configurations.  The problem was that the page size selector was only looking at one of those contexts.  So i created a prop we can pass in from each table which fixes the issue.  But I still dont like how we have the same context copy/pasted 5 times